File: Python/Problems/Problems/problem2186.py
import heapq
import math
import sys

import numpy as np

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter

from .problem import Problem

class Problem2186(Problem):
    def test(self) -> bool:
        s = "leetcode"
        t = "coats"

        expected = 7

        result = self.minSteps(s, t)

        print(result)

        return result == expected

    def minSteps(self, s: str, t: str) -> int:
        counts = [0] * 26
        for c in s:
            counts[ord(c) - ord('a')] += 1
        for c in t:
            counts[ord(c) - ord('a')] -= 1

        result = 0
        for i in range(26):
            result += abs(counts[i])
        return result

File: C++/Problems/Problems/problem_2186.cpp
#include "problem_2186.h"

bool problem_2186::test() {
    std::string s = "leetcode";
    std::string t = "coats";

    int expected = 7;

    auto result = minSteps(s, t);

    std::cout << result << std::endl;

    return result == expected;
}

int problem_2186::minSteps(std::string s, std::string t) {
    std::vector counts(26, 0);
    for (auto c : s)
        counts[c - 'a']++;
    for (auto c : t)
        counts[c - 'a']--;
    int result = 0;
    for (int i = 0; i < 26; ++i)
        result += std::abs(counts[i]);
    return result;
}
